src/sync/orderbook.py

```diff
@@ -0,0 +1,125 @@
+from src.models.order_rewards_schema import OrderRewards
+from src.models.tables import SyncTable
+from src.post.aws import AWSClient
+from src.sync.common import last_sync_block
+from src.sync.config import SyncConfig
+from src.sync.record_handler import RecordHandler
+from src.sync.upload_handler import UploadHandler
+
+log = set_log(__name__)
+
+
+class OrderbookDataHandler(
+    RecordHandler
+):  # pylint:disable=too-few-public-methods,too-many-arguments
+    """
+    This class is responsible for consuming new dune records and missing values from previous runs
+    it attempts to fetch content for them and filters them into "found" and "not found" as necessary
+    """
+
+    def __init__(
+        self,
+        file_manager: FileIO,
+        block_range: BlockRange,
+        sync_table: SyncTable,
+        config: SyncConfig,
+        data_list: list[dict[str, Any]],
+    ):
+        super().__init__(block_range, sync_table, config)
+        log.info(f"Handling {len(data_list)} new records")
+        self.file_manager = file_manager
+        self.data_list = data_list
+
+    def num_records(self) -> int:
+        return len(self.data_list)
+
+    def write_found_content(self) -> None:
+        self.file_manager.write_ndjson(data=self.data_list, name=self.content_filename)
+
+    def write_sync_data(self) -> None:
+        # Only write these if upload was successful.
+        self.file_manager.write_csv(
+            data=[{self.config.sync_column: str(self.block_range.block_to)}],
+            name=self.config.sync_file,
+        )
+
+
+def sync_order_rewards(
+    aws: AWSClient, fetcher: OrderbookFetcher, config: SyncConfig, dry_run: bool
+) -> None:
+    """Order Rewards Data Sync Logic"""
+    sync_table = SyncTable.ORDER_REWARDS
+    block_range = BlockRange(
+        block_from=last_sync_block(
+            aws,
+            table=sync_table,
+            genesis_block=15719994,  # First Recorded Order Reward block
+        ),
+        block_to=fetcher.get_latest_block(),
+    )
+    sync_orderbook_data(
+        aws,
+        block_range,
+        config,
+        dry_run,
+        sync_table=sync_table,
+        data_list=OrderRewards.from_pdf_to_dune_records(
+            fetcher.get_order_rewards(block_range)
+        ),
+    )
+
+
+def sync_batch_rewards(
+    aws: AWSClient, fetcher: OrderbookFetcher, config: SyncConfig, dry_run: bool
+) -> None:
+    """Batch Reward Sync Logic"""
+    sync_table = SyncTable.BATCH_REWARDS
+    block_range = BlockRange(
+        block_from=last_sync_block(
+            aws,
+            table=sync_table,
+            genesis_block=15719994,  # First Recorded Order Reward block
+        ),
+        block_to=fetcher.get_latest_block(),
+    )
+    sync_orderbook_data(
+        aws,
+        block_range,
+        config,
+        dry_run,
+        sync_table,
+        data_list=BatchRewards.from_pdf_to_dune_records(
+            fetcher.get_order_rewards(block_range)
+        ),
+    )
```

Comment:
In my opinion, there's a lot of duplicated code across these two methods. Maybe this can benefit from one more parameter to be either batch or orders? Although, readability and understandability will probably take a hit doing that.

What do you think?

Reply:
I have already tried quite excessively to split up the duplicated components of this code, but I could try again one more time. At this point, the only thing I can see that is left to deduplicate is the fetching of `block_range` and then maybe creating a general interface for these two methods 

- `BatchRewards.from_pdf_to_dune_records`
- `OrderRewards.from_pdf_to_dune_records`

In fact, looking here at this code you even helped me to spot a bug which I will now highlight as a suggested code change.

```suggestion
def sync_order_rewards(
    aws: AWSClient, fetcher: OrderbookFetcher, config: SyncConfig, dry_run: bool
) -> None:
    """Order Rewards Data Sync Logic"""
    sync_table = SyncTable.ORDER_REWARDS
    block_range = BlockRange(
        block_from=last_sync_block(
            aws,
            table=sync_table,
            genesis_block=15719994,  # First Recorded Order Reward block
        ),
        block_to=fetcher.get_latest_block(),
    )
    sync_orderbook_data(
        aws,
        block_range,
        config,
        dry_run,
        sync_table=sync_table,
        data_list=OrderRewards.from_pdf_to_dune_records(
            fetcher.get_order_rewards(block_range)
        ),
    )


def sync_batch_rewards(
    aws: AWSClient, fetcher: OrderbookFetcher, config: SyncConfig, dry_run: bool
) -> None:
    """Batch Reward Sync Logic"""
    sync_table = SyncTable.BATCH_REWARDS
    block_range = BlockRange(
        block_from=last_sync_block(
            aws,
            table=sync_table,
            genesis_block=15719994,  # First Recorded Order Reward block
        ),
        block_to=fetcher.get_latest_block(),
    )
    sync_orderbook_data(
        aws,
        block_range,
        config,
        dry_run,
        sync_table,
        data_list=BatchRewards.from_pdf_to_dune_records(
            fetcher.get_batch_rewards(block_range)
        ),
    )
```

Reply:
I think that's a great idea. And glad you found that, although I'm not quite sure what was replaced with what.

Reply:
RIght. I guess you can't see it because of the file rename. I will revert the file rename so you can see what really happened here. Sorry about that.

Reply:
Ok --- you will find all this stuff now under the original  file name `order_rewards.py` with a proper DIFF in the code.